### PR TITLE
Add fix for nodePort and clusterIp flaky tests

### PR DESCRIPTION
**Pull request by rishabhmsra**
Added fix for below nodeport and clusterIp related testcases:
- tests.v3_api.test_workload.test_wl_with_nodePort
- tests.v3_api.test_workload.test_wl_with_nodePort_scale_and_upgrade
- tests.v3_api.test_workload.test_wl_with_clusterIp
- tests.v3_api.test_workload.test_wl_with_clusterIp_scale_and_upgrade

### CODE REVIEW

**Comment by izaac** [comment]
@rishabhmsra need the branch to be rebased and a direct message me a link to the freeform test execution pointing to this branch with the needs_cluster or certification tests.

**Review by izaac — APPROVED** [review]
I've checked the test executions from this branch and it does look more stable.
There was one done with the complete certification test suite and one with only the related tests. 

Both passing.  The change is small but the analysis/debug was thorough.

LGTM

**Review by izaac — APPROVED** [review]

**Review by sgapanovich — APPROVED** [review]

**Comment by samkulkarni20** [comment]
@izaac you can go ahead & merge this PR, if everything is good. Me or @rishabhmsra can't merge any PRs yet.

**Pull request merged**
